### ISSUE TITLE
NecroVision and NecroVision: Lost Company does not work with DXVK

### ISSUE THREAD

**Issue by bear-bulloff** [doitsujin/dxvk#2299]
These games from GOG will quit on startup with DXVK but will work with wined3d, tested with DXVK 1.9.1 and 1.9.2.

System info:
GPU: RX 560 4gb
wine: 6.16 and 6.17
Mesa: 21.2.1

P.S. Read ProtonDB, that the game does not work with DX10 and need in config.ini file, line Cfg.RenderDevice = 2 replace 2 to 1, after these actions, the game started.
[necrovisiondxvk.log](https://github.com/doitsujin/dxvk/files/7203084/necrovisiondxvk.log)

**Comment by K0bin** [comment]
That's the logs. We also need apitraces.

**Comment by bear-bulloff** [comment]
> Это журналы. Нам также нужны апитрасы.
Either I'm doing something wrong or they're not being created.

**Comment by bear-bulloff** [comment]
![image](https://user-images.githubusercontent.com/59369672/134215401-5a04cef3-3159-4911-9931-0c1ca70585b7.png)
This is running the game without dxvk.
![Screenshot_367](https://user-images.githubusercontent.com/59369672/134215467-ef240d83-258e-4281-801c-6c3ef5af05e7.png)

**Comment by K0bin** [comment]
>  This is running the game without dxvk.

That's to be expected. The apitrace file should be in the desktop folder of your Wine prefix now.

**Comment by doitsujin** [comment]
Looks like the game requires the D3D10 effects framework, which we don't implement. Does `WINEDLLOVERRIDES=d3d10=b` fix this?

**Comment by bear-bulloff** [comment]
> > This is running the game without dxvk.
> 
> That's to be expected. The apitrace file should be in the desktop folder of your Wine prefix now.

After many different attempts at launching, it was possible to create one.
https://disk.yandex.ru/d/FeB4N0fjrsqwPw

**Comment by bear-bulloff** [comment]
> Looks like the game requires the D3D10 effects framework, which we don't implement. `WINEDLLOVERRIDES=d3d10=b`Does fix this?

This parameter does not correct

**Comment by doitsujin** [comment]
It's not a launch parameter, it's an environment variable. Please make sure you're applying it correctly (i.e. set Steam launch options to `WINEDLLOVERRIDES=d3d10=b %command%`).

**Comment by doitsujin** [comment]
ok, I tried this now and cannot get the game to work with Proton at all, even with wined3d. I managed to record an apitrace on Windows though which replays perfectly fine, so the problem is likely with some D3D10 utlity library (such as the effects framework). When trying to use DXVK on Windows, the game errors out with a message that it cannot create a D3D10 device for whatever reason, which is not what happens on Linux.

TL;DR no fucking clue what the game is doing or how to further debug this.

**Comment by bear-bulloff** [comment]
> ok, I tried this now and cannot get the game to work with Proton at all, even with wined3d. I managed to record an apitrace on Windows though which replays perfectly fine, so the problem is likely with some D3D10 utlity library (such as the effects framework). When trying to use DXVK on Windows, the game errors out with a message that it cannot create a D3D10 device for whatever reason, which is not what happens on Linux.
> 
> TL;DR no fucking clue what the game is doing or how to further debug this.

Tested on Mint 20.2 with Wine 6.17, OpenSUSE Tumbleweed with proton 6.17 of its build, Garuda with Wine 6.16, all works with wined3d, and with dxvk crashes either immediately or a bit later.

**Comment by jrugia** [comment]
Not sure if the case is relevant here but this game is 32bit and is known to run out of virtual address space.

Make sure to either patch the game's exe to support large address aware or a Wine/ Proton version that enables it.

Been a while since I plsyed this game but I can take a look tomorrow and report back.

**Comment by Blisto91** [comment]
Needs d3dx10 to run with dxvk.
Can be installed with winetricks
```
winetricks d3dx10
```

**Comment by Blisto91** [comment]
If any of you still have issues then just comment here again.
